### PR TITLE
net: lwm2m: fix registration content format and use plain/text

### PR DESCRIPTION
**Pull request by rbtchc**
According to the specification 5.3.1, it's a MUST to specify
(1) content format: app link format (2) supported lwm2m version.
Also, we should use text/plain instead of LWM2M's (obsolete).

Signed-off-by: Robert Chou <robert.ch.chou@acer.com>

### CODE REVIEW

File: subsys/net/lib/lwm2m/lwm2m_engine.h [hunk 1]
```diff
@@ -9,6 +9,22 @@
...
 
+#define LWM2M_PROTOCOL_VERSION "1.0"
+
+/* LWM2M / CoAP Content-Formats */
+#define LWM2M_FORMAT_PLAIN_TEXT		0
+#define LWM2M_FORMAT_APP_LINK_FORMAT	40
+#define LWM2M_FORMAT_APP_OCTET_STREAM	42
+#define LWM2M_FORMAT_APP_EXI		47
+#define LWM2M_FORMAT_APP_JSON		50
+#define LWM2M_FORMAT_OMA_PLAIN_TEXT	1541
```
Comment (mike-scott) [inline subsys/net/lib/lwm2m/lwm2m_engine.h]: If using LWM2M_FORMAT_OMA_PLAIN_TEXT is undesirable.  Should we remove the define entirely?

Comment (rbtchc) [inline subsys/net/lib/lwm2m/lwm2m_engine.h]: I did some research before, it seems that some legacy lwm2m implementations are still using the OMA_PLAIN_TEXT ID.
I think it's a decision that either we want to be compatible w/ the legacy or not.

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 2]
```diff
@@ -1973,17 +1961,16 @@ static int handle_request(struct zoap_packet *request,
...
 	/* read Accept */
 	r = zoap_find_options(in.in_zpkt, ZOAP_OPTION_ACCEPT, options, 1);
 	if (r > 0) {
 		accept = zoap_option_value_to_int(&options[0]);
 	} else {
-		SYS_LOG_DBG("No Accept header: use same as content-format(%d)",
-			    format);
-		accept = format;
+		SYS_LOG_DBG("No accept option given. Assume OMA TLV.");
+		accept = LWM2M_FORMAT_OMA_TLV;
```
Comment (mike-scott) [inline subsys/net/lib/lwm2m/lwm2m_engine.c]: This change seems out of place as it's not mentioned in the patch description.

File: subsys/net/lib/lwm2m/lwm2m_engine.h (hunk 1 above)
Comment (mike-scott) [inline subsys/net/lib/lwm2m/lwm2m_engine.h]: Ok sounds good.

File: subsys/net/lib/lwm2m/lwm2m_engine.c (hunk 2 above)
Comment (rbtchc) [inline subsys/net/lib/lwm2m/lwm2m_engine.c]: Update the commit description to explain the modification of choosing OMV TLV as a default accept format.

**Review by mike-scott — APPROVED** [review]
Thanks @rbtchc   LGTM.

Note: in the future you may consider breaking a patch like this up into 3 different patches: 1) to adjust the PLAIN_TEXT usage, 2) add the required parameters to the registration 3) change the default accept format.

**Comment by rbtchc** [comment]
got it.

**Review by jukkar — APPROVED** [review]

**Pull request merged**